DEVELOPER.md

```diff
@@ -2,380 +2,515 @@
+| `PATH_PROTOCOL`         | Sets the Azure Block Storage protocol.                                                          | `azb://`         |
+| `BUCKET_OR_CLUSTER`     | The account name.                                                                               | `account-name`   |
+| `PATH_PREFIX`           | Identifies the developer that created the database. Must begin and end with a slash (`/`).      | `/username/`     |
 
-         Env Name | Description | Sample value
-         | :--- | ---: | :---:
-         CONTAINERNAME  |Name of the azure container | container-name
-         ACCOUNTKEY  | If authenticating with an account key, fill this in.  Otherwise it can be left blank. | 
-         SHAREDACCESSSIGNATURE  | If authenticating with a SAS, fill this in.  This can be left blank if using an accountKey.  Before to include it in quotes ("") because of the special characters that are typically used. | 
-         PATH_PROTOCOL | Set this to tell the e2e tests that Azure is being used. | azb://
-         BUCKET_OR_CLUSTER  | Fill in the account name | account-name
-         PATH_PREFIX  | Include your user name here so that all dbs that we know who created the DBs.  It must begin and end with a slash. | /johndoe/
+### Pod logs
 
+The e2e tests use [stern](https://github.com/stern/stern) to persist some pod logs to help debug any failures. The e2e tests create the `int-tests-output/` directory to store the logs.
 
-5. Run the integration tests.
-   ```shell
-   kubectl kuttl test
-   ```
-   
-### Stern output
+The stern process completes when the kuttle tests run to completion. If you abort a kuttle test, then you must stop the stern process manually.
 
-The e2e tests use stern to save off logs of some pods.  This is done to aid in debugging any failures.  If needed, the logs are stored in the `int-tests-output` directory by default.  Cleanup of the stern process is only done if kuttl runs to completition.  If you abort the kuttl run, then you will need to stop the stern process manually.
+## Soak tests
 
-## 7. Running Soak Tests
+The soak tests evaluate the operator over a long interval. The test is split into multiple iterations, and each iteration generates a random workload that is comprised of pod kills and scaling operations. If the tests succeed, the next iteration begins. You can set the number of iterations that the soak test runs.
 
-The soak test will test the operator over a long interval. It splits the test into multiple iterations. Each iteration generates a random workload that is comprised of pod kills and scaling. At the end of each iteration, the test waits for everything to come up. If the test is successful, it proceeds to another iteration. It repeats this process for a set number of iterations or indefinitely.
+Soak tests are run with kuttl, and the random test generation is done with the [kuttle-step-gen tool](https://kuttl.dev/docs/testing/steps.html).
```

Comment:
The kuttl-step-gen is an internal tool developed on this repo. Might make sense to not link to kuttl. Maybe cmd/kuttl-step-gen?
```suggestion
Soak tests are run with kuttl, and the random test generation is done with the [kuttl-step-gen tool](cmd/kuttl-step-gen/main.go).
```